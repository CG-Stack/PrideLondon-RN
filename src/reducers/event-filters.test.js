// @flow
import reducer from "./event-filters";

describe("Event filters reducer", () => {
  it("initialises with default state", () => {
    // $FlowFixMe
    const state = reducer(undefined, {});

    expect(state).toMatchSnapshot();
  });

  it("updates state with filters from payload for UPDATE_EVENT_FILTERS action", () => {
    const initialState = {
      selectedFilters: {
        categories: new Set(),
        date: null,
        time: new Set()
      }
    };
    const state = reducer(initialState, {
      type: "UPDATE_EVENT_FILTERS",
      payload: {
        date: "2018-03-12"
      }
    });

    expect(state.selectedFilters.categories).toBe(
      initialState.selectedFilters.categories
    );
    expect(state.selectedFilters.date).toBe("2018-03-12");
    expect(state.selectedFilters.time).toBe(initialState.selectedFilters.time);
  });
});
