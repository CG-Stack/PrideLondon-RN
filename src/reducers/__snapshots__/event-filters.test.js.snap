// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event filters reducer initialises with default state 1`] = `
Object {
  "selectedFilters": Object {
    "categories": Set {},
    "date": null,
    "time": Set {},
  },
}
`;
